### PR TITLE
V22F-803 Intervention icon keeps showing

### PR DESCRIPTION
**Pull request by klieverse**
fix intervention icon showing when turned off, flashing steering wheel icon, and participant not able to accelerate when black box is braking in complete takeover

### CODE REVIEW

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]
Cannot steer, nor does the BB steer

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]
Decisions linger on through races

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionMakerTests.cpp [hunk 1]
```diff
@@ -14,13 +14,25 @@
 #include "portability.h"
 #include "RppUtils.hpp"
 #include "VariableStore.h"
+#include "Mediator.h"
+#include "Mediator.inl"
+#include "SDAConfig.h"
+#include "mocks/DecisionMakerMock.h"
+/// @brief A mediator that uses the standard SDecisionMakerMock
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionMakerTests.cpp:21]:
```suggestion

/// @brief A mediator that uses the standard SDecisionMakerMock
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/ExecutorTests.cpp [hunk 2]
```diff
@@ -2,11 +2,18 @@
...
 #include "SDAConfig.h"
-#include "mocks/DecisionMakerMock.h"
 #include "RppUtils.hpp"
 #include "mocks/DecisionMock.h"
 #include "IndicatorConfig.h"
+#include "Mediator.h"
+#include "Mediator.inl"
+#include "SDAConfig.h"
+#include "mocks/DecisionMakerMock.h"
+/// @brief A mediator that uses the standard SDecisionMakerMock
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/ExecutorTests.cpp:12]:
```suggestion

/// @brief A mediator that uses the standard SDecisionMakerMock
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MsgpackDeserializeTests.cpp [hunk 3]
```diff
@@ -7,15 +7,28 @@
 #include "DecisionTuple.h"
 #include "mocks/BlackBoxDataMock.h"
 
+#include "Mediator.h"
+#include "Mediator.inl"
+#include "SDAConfig.h"
+#include "mocks/DecisionMakerMock.h"
+/// @brief A mediator that uses the standard SDecisionMakerMock
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MsgpackDeserializeTests.cpp:14]:
```suggestion

/// @brief A mediator that uses the standard SDecisionMakerMock
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp [hunk 4]
```diff
@@ -255,11 +267,11 @@ TEST(RecorderTests, WriteDecisions)
...
-        decisionTuple.SetAccel(random.NextFloat(0, 1));
-        decisionTuple.SetBrake(random.NextFloat(0, 1));
-        decisionTuple.SetGear(random.NextInt(0, 10));
-        decisionTuple.SetSteer(random.NextFloat(0, 1));
-        expectedDecisionsData << bits(timestamp) << bits(decisionTuple.GetSteer()) << bits(decisionTuple.GetAccel()) << bits(decisionTuple.GetBrake()) << bits(static_cast<float>(decisionTuple.GetGear()));
+        decisionTuple.SetAccelDecision(random.NextFloat(0, 1));
+        decisionTuple.SetBrakeDecision(random.NextFloat(0, 1));
+        decisionTuple.SetGearDecision(random.NextInt(0, 10));
+        decisionTuple.SetSteerDecision(random.NextFloat(0, 1));
+        expectedDecisionsData << bits(timestamp) << bits(decisionTuple.GetSteerAmount()) << bits(decisionTuple.GetAccelAmount()) << bits(decisionTuple.GetBrakeAmount()) << bits(static_cast<float>(decisionTuple.GetGearAmount()));
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp:274]: This line is way too long, spread it over multiple lines, for example every `<<`

**Review by larsdekwant — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/CarController.cpp [hunk 5]
```diff
@@ -8,37 +8,27 @@
...
-{
-    m_thresholds = p_thresholds;
 }
 
 /// @brief			 Edits the clutch command of the game
 /// @param  p_clutch The amount that needs to be edited
 void CarController::SetClutchCmd(float p_clutch) const
 {
-    m_car->ctrl.clutchCmd = p_clutch;
+    m_car->ctrl.clutchCmd += p_clutch;
```
[the remark this answers is not in the document]
Comment (klieverse) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/CarController.cpp:31]: The decision maker can send +1, 0 or -1, hence it should be added. But, clutch and gear will be removed in a PR in the near future

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp (hunk 4 above)
Comment (klieverse) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp:274]: Whenever I do that the code coverage tool fails on Linux (but not on Windows). I would like some further instructions on how I could solve that.

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Decision.h [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+/// @brief An interface for a decision that can be made by an AI
+class Decision
+{
+public:
+    /// @brief Can show an indicator
+    void RunIndicateCommands();
+
+    /// @brief Intervenes the simulation if allowed to run the command
+    void RunInterveneCommands(tAllowedActions p_allowedActions);
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Decision.h:12]: missing @param in documentation.

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Decision.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+    /// @brief Sets the intervention amount and checks whether the threshold is reached
+    void SetInterventionAmount(float p_interventionAmount);
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Decision.h:15]: missing @param in documentation.

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl [hunk 8]
```diff
@@ -176,3 +179,15 @@ Recorder* TEMP_DECISIONMAKER::GetRecorder()
 {
     return m_recorder;
 }
+
+template <typename SocketBlackBox, typename SDAConfig, typename FileDataStorage, typename SQLDatabaseStorage, typename Recorder>
+DecisionTuple TEMP_DECISIONMAKER::GetDecisions()
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl:184]: Missing documentation, I also see that documentation misses on the 4 functions above, could you also add that? Appereantly it didn't get noticed in those PRs.

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl [hunk 9]
```diff
@@ -176,3 +179,15 @@ Recorder* TEMP_DECISIONMAKER::GetRecorder()
...
+{
+    return m_decision;
+}
+
+template <typename SocketBlackBox, typename SDAConfig, typename FileDataStorage, typename SQLDatabaseStorage, typename Recorder>
+void TEMP_DECISIONMAKER::SetDecisions(DecisionTuple p_decision)
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl:190]: Missing documentation

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp (hunk 4 above)
Comment (larsdekwant) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp:274]: Something like this should work, if that doesn't you can just add the expectedDecisionsData stream in front of every `<<`.
```suggestion
        expectedDecisionsData << bits(timestamp)
                              << bits(decisionTuple.GetSteerAmount())
                              << bits(decisionTuple.GetAccelAmount())
                              << bits(decisionTuple.GetBrakeAmount())
                              << bits(static_cast<float>(decisionTuple.GetGearAmount()));
```

**Review by larsdekwant — CHANGES_REQUESTED** [review]

**Review by larsdekwant — CHANGES_REQUESTED** [review]
Remove the intervention type logging in `InterventionFactory.cpp`, both the comment and the std::cout.
It's in the CASE define:
- `// THE STD::COUT LINE CAN BE REMOVED AFTER THE DEMO (IF WE DO ONE)`
- `std::cout << #type << std::endl;`

**Review by larsdekwant — COMMENTED** [review]
On hold until the interventions are testable again with a working blackbox

**Review by suppergerrie2 — APPROVED** [review]

**Review by larsdekwant — APPROVED** [review]

**Review by noutheesink — APPROVED** [review]

**Pull request merged**